File: Raven.Client.Lightweight/Listeners/IDocumentQueryListener.cs
namespace Raven.Client.Listeners
{
	/// <summary>
	/// Hook for users to modify all queries globally
	/// </summary>
	public interface IDocumentQueryListener
	{
		/// <summary>
		/// Allow to customize a query globally
		/// </summary>
		void BeforeQueryExecuted(IDocumentQueryCustomization queryCustomization);
	}
}

File: Raven.Client.Java/lib/src/main/java/raven/client/listeners/IDocumentQueryListener.java
package raven.client.listeners;

import raven.client.IDocumentQueryCustomization;

/**
 * Hook for users to modify all queries globally
 */
public interface IDocumentQueryListener {
  /**
   * Allow to customize a query globally
   * @param queryCustomization
   */
  void beforeQueryExecuted(IDocumentQueryCustomization queryCustomization);
}
